app/src/main/java/bean/pwr/imskamieskiego/MapActivity.java

```diff
@@ -327,4 +309,67 @@ public void startPointSearchRequest() {
     public void startNavigation(boolean avoidStairs) {
         Log.i(TAG, "startNavigation: start navigation");
     }
+
+    private void floorListPopupWindowInit() {
+
+
+        floorListPopupWindow = new PopupWindow(MapActivity.this);
+        View popupLayout = getLayoutInflater().inflate(R.layout.choose_floor_list, null);
+        ListView floorListView = popupLayout.findViewById(R.id.floor_list);
+        floorListView.setAdapter(floorsAdapter());
+        floorListPopupWindow.setContentView(popupLayout);
+
+
+        floorListView.setOnItemClickListener(
+
+                (parent, view, position, id) -> {
+                    if (currentFloor != position) {
+                        currentFloor = position;
+                        floorListPopupWindow.dismiss();
+                        floorViewModel.setCurrentFloor(currentFloor);
+                        floorViewModel.setSelectedFloor(currentFloor);
+                        changeFloorButton.setText(Integer.toString(currentFloor));
+
+                    }
+                });
+
+
+        floorListPopupWindow.setHeight(WindowManager.LayoutParams.WRAP_CONTENT);
+        floorListPopupWindow.setWidth(WindowManager.LayoutParams.WRAP_CONTENT);
+        floorListPopupWindow.setOutsideTouchable(true);
+        floorListPopupWindow.setFocusable(true);
+
+    }
+
+
+    ArrayAdapter<String> floorsAdapter() {
+        ArrayList<String> floors = new ArrayList<>();
+
+        floorViewModel.getFloorList().observe(MapActivity.this, floorList -> {
+            if (floorList != null) {
+
+                Collections.addAll(floors, floorList);
+            }
+        });
+
+        return new ArrayAdapter<String>(MapActivity.this, android.R.layout.simple_list_item_1, floors) {
+
+            @NonNull
+            @Override
+            public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
+                View returnedView = super.getView(position, convertView, parent);
+
+                if (position == currentFloor) {
+                    returnedView.setBackgroundColor(getColor(R.color.colorPrimary));
```

Comment:
Takie wywołanie jakie Ty zrobiłaś działa od API 23.
```suggestion
                    returnedView.setBackgroundColor(getResources().getColor(R.color.colorPrimary));
```